core/src/main/java/com/github/kaktushose/jda/commands/embeds/EmbedDataSource.java

```diff
@@ -0,0 +1,90 @@
+package com.github.kaktushose.jda.commands.embeds;
+
+import com.github.kaktushose.jda.commands.JDACBuilder.ConfigurationException;
+import com.github.kaktushose.jda.commands.i18n.I18n;
+import net.dv8tion.jda.api.exceptions.ParsingException;
+import net.dv8tion.jda.api.utils.data.DataObject;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Optional;
+
+/// An [EmbedDataSource] is used to retrieve [Embed]s based on a unique name from various sources.
+@FunctionalInterface
+public interface EmbedDataSource {
+
+    /// Constructs a new [EmbedDataSource] using a JSON payload as its source.
+    ///
+    /// @param json the JSON payload to retrieve embeds from
+    /// @return a new [EmbedDataSource]
+    @NotNull
+    static EmbedDataSource json(@NotNull String json) {
+        return dataObject(DataObject.fromJson(json));
+    }
+
+    /// Constructs a new [EmbedDataSource] using a JSON file that is located inside the resources folder.
+    ///
+    /// @param resource the [Path] pointing to a JSON file
+    /// @return a new [EmbedDataSource]
+    @NotNull
+    static EmbedDataSource resource(@NotNull String resource) {
+        try (InputStream inputStream = EmbedDataSource.class.getClassLoader().getResourceAsStream(resource)) {
+            if (inputStream == null) {
+                throw new ConfigurationException("Failed to find resource %s".formatted(resource));
+            }
+            return inputStream(inputStream);
+        } catch (IOException e) {
+            throw new ConfigurationException("Failed to open file", e);
+        }
+    }
+
+    /// Constructs a new [EmbedDataSource] using a [Path] pointing to a JSON file as its source.
+    ///
+    /// @param path the [Path] pointing to a JSON file
+    /// @return a new [EmbedDataSource]
+    @NotNull
+    static EmbedDataSource file(@NotNull Path path) {
+        try {
+            return inputStream(Files.newInputStream(path));
```

Comment:
I would prefer using `json(Files.readString(path))` as it's more "direct" than using an input stream.